test/boost/schema_change_test.cc

```diff
@@ -75,6 +80,141 @@ SEASTAR_TEST_CASE(test_new_schema_with_no_structural_change_is_propagated) {
+                BOOST_REQUIRE(!create_statement.contains("$test_set_c2"));
+            }
+
+            // Check internal columns are invisible for wildcard selects.
+            {
+                const auto cql = format("select * from tests_ks.{}", table_name);
+                eventually([&] {
+                    const auto msg = e.execute_cql(cql).get();
+                    assert_that(msg).is_rows().with_size(1).with_row({
+                        {int32_type->decompose(1)}, 
+                        {int32_type->decompose(2)}, 
+                        {int32_type->decompose(3)}
+                    });
+                });
+            }
+
+            // Check internal columns can be accessed by name.
+            {
+                const auto cql = format("select \"$test_set_c1\", \"$test_set_c2\" from tests_ks.{}", table_name);
+                eventually([&] {
+                    const auto msg = e.execute_cql(cql).get();
+                    assert_that(msg).is_rows().with_size(1).is_null();
+                });
+            }
+            if (original_schema->is_view()) {
+                const auto ts = api::new_timestamp();
+                mutation m(updated_schema, partition_key::from_singular(*updated_schema, 1));
+                m.set_clustered_cell(clustering_key{}, "$test_set_c1", static_cast<int32_t>(10), ts);
+                m.set_static_cell("$test_set_c2", "test-str", ts);
+                e.get_storage_proxy().local().mutate_locally(updated_schema, 
+                    freeze(m), nullptr, db::commitlog::force_sync::no).get();
+            } else {
+                const auto cql = format("update tests_ks.{} "
+                    "set \"$test_set_c1\" = 10, \"$test_set_c2\" = 'test-str' "
+                    "where pk = 1", table_name);
+                e.execute_cql(cql).get();
+            }
+            {
+                const auto cql = format("select \"$test_set_c1\", \"$test_set_c2\" from tests_ks.{}", table_name);
+                const auto msg = e.execute_cql(cql).get();
+                assert_that(msg).is_rows().with_size(1).with_row({
+                    {int32_type->decompose(10)}, 
+                    {utf8_type->decompose("test-str")}
+                });
+            }
+        };
+
+        e.execute_cql("create keyspace tests_ks with replication = { 'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1 };").get();
+        e.execute_cql("create table tests_ks.table1 (pk int primary key, c1 int, c2 int);").get();
+        e.execute_cql("insert into tests_ks.table1 (pk, c1, c2) values (1, 2, 3)").get();
+
+        // Check can inject internal columns into a table.
+        check_inject_into("table1");
+
+        // Check can inject internal columns into a view.
+        e.execute_cql("create materialized view tests_ks.view1 as select pk, c1, c2 from tests_ks.table1 "
+            "where pk is not null "
+            "primary key (pk)").get();
+        check_inject_into("view1");
```

Comment:
I would like some checks for `select * from table` with internal columns not showing the internal columns and `select $internal_column from table` showing its value.

Reply:
Also, a check would be useful to make sure the view didn't consider the internal column.

Reply:
> I would like some checks for select * from table with internal columns not showing the internal columns and select $internal_column from table showing its value.

That's exactly the kind of checks the `check_inject_into` contains. We run it for table and view.

Reply:
> Also, a check would be useful to make sure the view didn't consider the internal column.

Added the test.